packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -14,44 +14,47 @@ import { createSigner, fetchSignerType, loadSignerType, storeSignerType } from '
 import { checkErrorForInvalidRequestArgs } from './util/provider';
 import { Communicator } from ':core/communicator/Communicator';
 import { SignerType } from ':core/message';
+import { clearAllStorage } from ':core/storage/util';
 import { hexStringFromNumber } from ':core/type/util';
-import type { BaseStorage } from ':util/BaseStorage';
-import { ScopedStorage } from ':util/ScopedStorage';
 
 export class CoinbaseWalletProvider extends EventEmitter implements ProviderInterface {
   private readonly metadata: AppMetadata;
   private readonly preference: Preference;
   private readonly communicator: Communicator;
-  private readonly baseStorage?: BaseStorage;
 
-  private signer: Signer | null;
+  private initPromise: Promise<void>;
+  private signer: Signer | null = null;
 
-  constructor({
-    baseStorage,
-    metadata,
-    preference: { keysUrl, ...preference },
-  }: Readonly<ConstructorOptions>) {
+  constructor({ metadata, preference: { keysUrl, ...preference } }: Readonly<ConstructorOptions>) {
     super();
-    this.baseStorage = baseStorage;
     this.metadata = metadata;
     this.preference = preference;
     this.communicator = new Communicator(keysUrl);
+
+    // Async initialize
+    this.initPromise = this.initialize();
```

Comment:
ahhh pretty good solution this problem, I have another solution that possibly can get rid of the initPromise, which i think can be not so intuitive? wonder wdyt

```suggestion
  private constructor({
    metadata,
    preference: { keysUrl, ...preference },
  }: Readonly<ConstructorOptions>) {
    super();
    this.metadata = metadata;
    this.preference = preference;
    this.communicator = new Communicator(keysUrl);
  }

  private async initialize() {
    // Load states from storage
    const signerType = await loadSignerType();
    if (signerType) {
      this.signer = await this.initSigner(signerType);
    }
  }

  public static async createInstance(
    options: Readonly<ConstructorOptions>
  ): Promise<CoinbaseWalletProvider> {
    const instance = new CoinbaseWalletProvider(options);
    await instance.initialize();
    return instance;
  }
```

Reply:
@fan-zhang-sv I do like the `static async createInstance` method better, but it requires changing the public interface of `CoinbaseWalletSDK.makeWeb3Provider` to be async. I'm hesitant to do this since may break apps that rely on this to be synchronous. 

I went ahead and made the change for the `SCWSigner` class though.